rfc/plugins.md

```diff
@@ -0,0 +1,300 @@
+
+The proposed plugin system will aim for feature parity of the existing plugin system, but will not guarantee backwards compatibility of developer or operator experience. The priority is in satisfying broad-strokes business requirements, as opposed to satisfying existing technical capabilities.
+
+## Known Unknowns
+There are a lot of known unknowns. It’s the intention that separate proposals will be made for many of these unknowns, such as development and publishing lifecycle, as well as operator experiences.
+
+* In-process plugins
+	* How will plugin authors get metrics, logs, or crash reports from their plugins?
+	* How will operators be notified that new plugin versions exist and can be released?
+	* Do we want to support updating backend plugins at runtime?
+* Deck plugins
+	* We want to experiment with some Webpack magic to achieve plugin loading, although we’re unsure if this will be possible yet.
+
+_What parts of the design do you expect to resolve through the RFC process?_
+_What parts of the design do you expect to resolve through implementation before stabilization?_
+_What related issues do you consider out of scope for this RFC that could be addressed in the future?_
+
+## Security, Privacy, and Compliance
+In-process plugins present an interesting security issue. Plugins can introduce new dependencies which expose CVEs, they can access internal APIs that may break core—or other plugins—behavior. Further, a story around metrics, logs and error reporting for plugin developers becomes harder as there’s greater possibility for leaking potentially sensitive information to parties that do not necessarily have organizational approval for such data.
+
+Initially speaking, things like logs, metrics and crash reports for in-process plugins will go unaddressed, putting the sole burden of security judgment on the Operator.
+
+In-process plugins are not fully isolated, so close scrutiny of a plugin’s behavior is mandatory for any Operator prior to installation or enabling.
+
+## Operations
+This proposal alters how Spinnaker development will be performed, as well as how Spinnaker functionality is released and delivered to various environments. We do not yet know the exact ways that this will change, however the goal is to make development of Spinnaker—for both core and extensions—easier.
+
+## Risks
+* There is an existing plugin system that has made assumptions on plugin loading. There is risk that switching its internal functionality to PF4J may be in some ways backwards incompatible; such as how plugins are detected, or how config is loaded for them.
+	* This will likely cause breakages in existing plugins that will need to be migrated.
```

Comment:
Do we have an idea how many plugins have been developed against the old system?

Reply:
A very small amount. It's not something to lose sleep over, but this number will increase the longer we wait to provide a longer-term solution.